octopoes/bits/ssl_certificate_hostname/ssl_certificate_hostname.py

```diff
@@ -52,7 +52,7 @@ def run(
     for website in websites:
         hostname = website.hostname.tokenized.name.rstrip(".")
 
-        if subject_valid_for_hostname(subject, hostname):
+        if subject is not None and subject_valid_for_hostname(subject, hostname):
```

Comment:
Our validation currently doesn't match the method specified in [RFC2818](https://datatracker.ietf.org/doc/html/rfc2818#section-3.1), which specifies that only `subjectAltName` should be checked when it is present:

> If a subjectAltName extension of type dNSName is present, that MUST
> be used as the identity. Otherwise, the (most specific) Common Name
> field in the Subject field of the certificate MUST be used. Although
> the use of the Common Name is existing practice, it is deprecated and
> Certification Authorities are encouraged to use the dNSName instead. 

Modern browsers remove validation of the Common Name field entirely, e.g. [Chrome removed support in 2017](https://bugs.chromium.org/p/chromium/issues/detail?id=308330).

Reply:
Agree.